### PR TITLE
Add text customization for pay button

### PR DESCRIPTION
**Pull request by Kukks**
https://i.imgur.com/nFxscOZ.gifv

### CODE REVIEW

**Comment by pavlenex** [comment]
tACK. Not a fan of a checkbox for this,I'd rather have it as a regular field, but Kukks said it's needed.

File: BTCPayServer/wwwroot/paybutton/paybutton.js [hunk 1]
```diff
@@ -103,8 +107,15 @@ function inputChanges(event, buttonSize) {
...
-
-    html += '  <input type="image" class="submit" name="submit" src="' + esc(srvModel.payButtonImageUrl) + '" style="width:' + width + '" alt="Pay with BtcPay, Self-Hosted Bitcoin Payment Processor">\n';
+    
+    if(!srvModel.payButtonText){
+        html += '  <input type="image" class="submit" name="submit" src="' + esc(srvModel.payButtonImageUrl) + '" style="width:' + width + '" alt="Pay with BtcPay, Self-Hosted Bitcoin Payment Processor">\n';
+    }else{
+        var numwidth = parseInt(width.replace("px", ""));
+        html+= '<button type="submit" class="submit" name="submit" style="min-width:' + width + '; min-height:' + height + '; border-radius: 4px;border-style: none;background-color: #0f3b21;" alt="Pay with BtcPay, Self-Hosted Bitcoin Payment Processor"><span style="color:#fff">'+srvModel.payButtonText+'</span>\n' +
+            (srvModel.payButtonImageUrl? '<img src="'+srvModel.payButtonImageUrl+'" style="width:'+numwidth/2+'px;">\n' : '')+
+            '</button>'
```
Comment (NicolasDorier) [inline BTCPayServer/wwwroot/paybutton/paybutton.js]: need escapes

File: BTCPayServer/Views/Stores/PayButton.cshtml [hunk 2]
```diff
@@ -232,13 +243,33 @@
...
+                
+            },
+            watch: {
+                buttonInlineTextMode: function (checked) {
+                  if(!checked){
+                      this.srvModel.payButtonText = "";
+                      this.srvModel.payButtonImageUrl = this.originalButtonImageUrl;
+                  }else {
+                      this.srvModel.payButtonText = "Pay with..";
+                      this.srvModel.payButtonImageUrl = this.srvModel.urlRoot + "img/logo.svg";
```
Comment (NicolasDorier) [inline BTCPayServer/Views/Stores/PayButton.cshtml]: escape

File: BTCPayServer/Views/Stores/PayButton.cshtml [hunk 3]
```diff
@@ -232,13 +243,33 @@
         var payButtonCtrl = new Vue({
             el: '#payButtonCtrl',
             data: {
-                srvModel: srvModel
+                srvModel: srvModel,
+                originalButtonImageUrl: srvModel.payButtonImageUrl+"",
```
Comment (NicolasDorier) [inline BTCPayServer/Views/Stores/PayButton.cshtml]: why `+ ""`

Comment (Kukks) [reply BTCPayServer/Views/Stores/PayButton.cshtml]: this one is fine, still in the page model, escapes only need to be the paybutton.js

Comment (Kukks) [reply BTCPayServer/Views/Stores/PayButton.cshtml]: just an old trickt to break a js var reference, but not needed

File: BTCPayServer/Views/Stores/PayButton.cshtml (hunk 2 above)
Comment (Kukks) [reply BTCPayServer/Views/Stores/PayButton.cshtml]: not needed in here as this is just the view model code

**Comment by Kukks** [comment]
updated PR @NicolasDorier

File: BTCPayServer/Views/Stores/PayButton.cshtml (hunk 3 above)
Comment (NicolasDorier) [reply BTCPayServer/Views/Stores/PayButton.cshtml]: wat?

Comment (Kukks) [reply BTCPayServer/Views/Stores/PayButton.cshtml]: removed it :P

**Pull request merged**